src/app/cmdoptions.cpp

```diff
@@ -497,7 +498,7 @@ QString wrapText(const QString &text, int initialIndentation = USAGE_TEXT_COLUMN
     QStringList lines = {words.first()};
     int currentLineMaxLength = wrapAtColumn - initialIndentation;
 
-    foreach (const QString &word, words.mid(1)) {
+    for (const QString &word : copyAsConst(words.mid(1))) {
```

Comment:
I guess you won't need `copyAsConst()` here, also you could use [clazy](https://github.com/KDE/clazy) to help verify.

Reply:
This is one of the 4 warnings Clazy complains.

Reply:
As for this change (and many other `copyAsConst` usages):
Since `mid()` doesn't actually return shared copy of some container `detach()` will not do anything except checking for references count so it's more lightweight than copy constructor called from `copyAsConst()`.

Reply:
I have no idea how you know what you said and consequently, how to spot the other copyAsConst that aren't needed.

Reply:
As a rule, it obviously stems from the meaning of the function.
E.g. `mid()` here creates new list containing some subset of elements of affected list.
Just try to examine the meaning of all other affected functions. If you're in doubt in a particular case, ask for my explanation.

The problem of "detach" is greatly exaggerated. When we began to use range based for loop, we saw a warning that it can cause "detach()". And we blindly rushed to solve it, without going into details. But now it is obvious to me that in most cases this is not a problem at all.
@Chocobo1, don't you mind join this discussion?

Reply:
There are a few existing copyAsConst that seem to return a new container e.g. https://github.com/qbittorrent/qBittorrent/blob/master/src/gui/ipsubnetwhitelistoptionsdialog.cpp#L49, https://github.com/qbittorrent/qBittorrent/blob/master/src/base/bittorrent/tracker.cpp#L139 and https://github.com/qbittorrent/qBittorrent/blob/master/src/webui/api/appcontroller.cpp#L205
Are they correct or do i remove them?
There is a case that returns a container that is returned from another function, it should keep copyAsConst right? https://github.com/qbittorrent/qBittorrent/blob/master/src/gui/properties/trackerlistwidget.cpp#L397

Reply:
If this isn't much work, it might be worth if we can make an non-exhaustive whitelist of situations (and put in coding guidelines) while reviewing this PR, specifying where `copyAsConst()` can be omitted. For example: `mid()` function in Qt types.

Reply:
>There are a few existing copyAsConst that seem to return a new container...
Are they correct or do i remove them?

It should be removed in all 3 cases above.
>There is a case that returns a container that is returned from another function, it should keep copyAsConst right?

No in case if first function returns no copy of existing object (as in your example).

In General, I came to the conclusion that the methods that return a newly created container (a copy of a temporary object, to be exact) are the vast majority. I do not even come to mind the opposite case (at least in Qt), it needs a good look. As for qBittorrent own code, the best solution would be to change the type of those few methods that return a shared copy of the class internal container so that they return a constant reference instead (e.g. https://github.com/qbittorrent/qBittorrent/pull/9883/files#diff-b369e532eff50253387c9d203da4b928R453) and remove `copyAsConst` at all so it did not tempt us. If we still encounter the affected method somewhere in the third-party code (or in Qt), we can always store its value in a pre-declared constant and iterate over it.

Reply:
>we can always store its value in a pre-declared constant and iterate over it.

I thought copyAsConst was introduced to avoid that? https://github.com/qbittorrent/qBittorrent/pull/8325#discussion_r164265321